automated install/basic-install.sh

```diff
@@ -351,40 +351,51 @@ package_manager_detect() {
-        if ! grep -qiE 'fedora|fedberry' /etc/redhat-release; then
-            if rpm -qa | grep -qi 'epel'; then
-                printf "  %b EPEL repository already installed\\n" "${TICK}"
-            else
-                local RH_RELEASE EPEL_PKG
-                # EPEL not already installed, add it based on the release version
-                RH_RELEASE=$(grep -oP '(?<= )[0-9]+(?=\.?)' /etc/redhat-release)
-                EPEL_PKG="https://dl.fedoraproject.org/pub/epel/epel-release-latest-${RH_RELEASE}.noarch.rpm"
-                printf "  %b Enabling EPEL package repository (https://fedoraproject.org/wiki/EPEL)\\n" "${INFO}"
-                "${PKG_INSTALL[@]}" "${EPEL_PKG}"
-                printf "  %b Installed %s\\n" "${TICK}" "${EPEL_PKG}"
-            fi
-        fi
+        if [ "${PKG_MANAGER}" = "dnf" ] || [ "${PKG_MANAGER}" = "yum" ] ; then
+          # CentOS package manager returns 100 when there are packages to update so we need to || true to prevent the script from exiting.
+          PKG_COUNT="${PKG_MANAGER} check-update | egrep '(.i686|.x86|.noarch|.arm|.src)' | wc -l || true"
+          INSTALLER_DEPS=(git dialog iproute newt procps-ng which chkconfig ca-certificates)
+          PIHOLE_DEPS=(cronie curl findutils sudo unzip libidn2 psmisc libcap nmap-ncat jq)
+          PIHOLE_WEB_DEPS=(lighttpd lighttpd-fastcgi php-common php-cli php-pdo php-xml php-json php-intl)
 
-    # If neither apt-get or yum/dnf package managers were found
+        # If the host OS is centos (or a derivative), epel is required for lighttpd
+          if ! grep -qiE 'fedora|fedberry' /etc/redhat-release; then
+              if rpm -qa | grep -qi 'epel'; then
+                  printf "  %b EPEL repository already installed\\n" "${TICK}"
+              else
+                  local RH_RELEASE EPEL_PKG
+                  # EPEL not already installed, add it based on the release version
+                  RH_RELEASE=$(grep -oP '(?<= )[0-9]+(?=\.?)' /etc/redhat-release)
+                  EPEL_PKG="https://dl.fedoraproject.org/pub/epel/epel-release-latest-${RH_RELEASE}.noarch.rpm"
+                  printf "  %b Enabling EPEL package repository (https://fedoraproject.org/wiki/EPEL)\\n" "${INFO}"
+                  "${PKG_INSTALL[@]}" "${EPEL_PKG}"
+                  printf "  %b Installed %s\\n" "${TICK}" "${EPEL_PKG}"
+              fi
+          fi
+        else
+          # openSUSE packages
+          # openSUSE package manager returns number when there are packages to update so we need to || true to prevent the script from exiting.
+          PKG_COUNT="${PKG_MANAGER} lu | egrep '(.i686|.x86|.noarch|.arm|.src)' | wc -l || true"
+          INSTALLER_DEPS=(git dialog iproute2 newt procps which ca-certificates)
+          PIHOLE_DEPS=(cronie curl findutils sudo unzip libidn2 psmisc libcap-ng0 netcat-openbsd jq)
+          PIHOLE_WEB_DEPS=(lighttpd php8 php8-fastcgi php8-cli php8-pdo php8-intl php8-openssl php8-sqlite)
```

Comment:
Are you sure there are PHP8 packages to openSUSE Leap?

Reply:
```
The following 25 NEW packages are going to be installed:\n  cron cronie jq libbrotlienc1 libbsd0 libdbi3 libicu-suse65_1 libicu65_1-ledata libjq1 libonig4 lighttpd mailx netcat-openbsd php8 php8-cli php8-fastcgi php8-intl php8-openssl php8-pdo php8-sqlite 
```

This part was just copied from https://github.com/pi-hole/pi-hole/pull/5001

Reply:
https://software.opensuse.org/package/php8?locale=en

> php8
> 
> Interpreter for the PHP scripting language version 8
> 
> PHP is a server-side HTML embedded scripting language designed primarily for web development but also used as a general-purpose programming language. This package contains the base files for all subpackages and must be installed in order to use PHP. Additionally, extension modules and server modules (e.g. for Apache) may be installed.
> 
> There is no official package available for openSUSE Leap 15.4

Reply:
Interesting, my local container says otherwise

```
1c9a8e950408:/ # php -v
PHP 8.0.25 (cli) (built: Oct 31 2022 12:00:00) ( NTS )
Copyright (c) The PHP Group
Zend Engine v4.0.25, Copyright (c) Zend Technologies
```

Reply:
What image are you using? What architecture?
Maybe there is a package difference between different architectures.

Reply:
`FROM opensuse/leap:15.4`

It's a x86_64 system

Reply:
I'm testing on a Raspberry Pi. Maybe the difference is there.

Reply:
I'm not sure if there will be someone running openSUSE on a Raspberry Pi.